### PR TITLE
Modify gatewayRESTAPI parameters 

### PR DESCRIPTION
**Pull request by Sarangan0219**
### Purpose
Related to wso2/product-apim#8246

### Approach

- Provide the ability to to access and modify synapse resources using  API Name , version and Tenant Domain.
- Changed the REST API parameters from` apiId` and `gatewayLabel`  to  `apiName`, `version` and `tenantDomain` for the resources of 

        1.  /deploy-api 
        2.  /undeploy-api
        3.  /api-artifact
        4.  /sequence
        5.  /local-entry
        6.  /end-points

- `apiName  `and `version` are required parameters. If the `tenantDomain `is not given **carbon.super** will be picked as the tenant domain.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml [hunk 1]
```diff
@@ -152,13 +156,17 @@ paths:
           in: query
           type: string
           required: true
-        - name: label
+        - name: version
           description: |
-            Label of the API Gateway
+            version of the API Gateway
```
Comment (1akshitha) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml:161]: This should be the version of the API

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/DeployApiApiServiceImpl.java [hunk 2]
```diff
@@ -19,19 +19,29 @@
...
 public class DeployApiApiServiceImpl implements DeployApiApiService {
+    private final String SUPER_TENAT_DOMAIN = "carbon.super";
 
-    public Response deployApiPost(String apiName, String gatewayLabel, String apiId, MessageContext messageContext) {
+    public Response deployApiPost(String apiName, String version , String tenantDomain, MessageContext messageContext) {
 
         InMemoryAPIDeployer inMemoryApiDeployer = new InMemoryAPIDeployer();
-        boolean status = inMemoryApiDeployer.deployAPI(apiId, gatewayLabel);
+        if (tenantDomain == null){
+            tenantDomain =SUPER_TENAT_DOMAIN;
```
Comment (1akshitha) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/DeployApiApiServiceImpl.java:39]: Formatting error

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/EndPointsApiServiceImpl.java [hunk 3]
```diff
@@ -26,22 +26,34 @@
...
 
     private static final Log log = LogFactory.getLog(EndPointsApiServiceImpl.class);
+    private final String SUPER_TENAT_DOMAIN = "carbon.super";
 
-    public Response endPointsGet(String apiName, String label, String apiId, MessageContext messageContext) {
+    public Response endPointsGet(String apiName, String version , String tenantDomain, MessageContext messageContext) {
 
         InMemoryAPIDeployer inMemoryApiDeployer = new InMemoryAPIDeployer();
+        if (tenantDomain == null){
+            tenantDomain =SUPER_TENAT_DOMAIN;
```
Comment (1akshitha) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/EndPointsApiServiceImpl.java:47]: formatting error

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/LocalEntryApiServiceImpl.java [hunk 4]
```diff
@@ -28,19 +28,29 @@
...
 import org.apache.cxf.jaxrs.ext.MessageContext;
 import org.wso2.carbon.apimgt.rest.api.util.utils.RestApiUtil;
 
 import javax.ws.rs.core.Response;
+import java.util.Map;
 
 public class LocalEntryApiServiceImpl implements LocalEntryApiService {
 
     private static final Log log = LogFactory.getLog(LocalEntryApiServiceImpl.class);
+    private final String SUPER_TENAT_DOMAIN = "carbon.super";
```
Comment (1akshitha) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/LocalEntryApiServiceImpl.java:42]: Don't we have this super tenant as a constant in APIConstants? If so please point to that.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/SequenceApiServiceImpl.java [hunk 5]
```diff
@@ -27,22 +27,31 @@
...
 import org.apache.cxf.jaxrs.ext.MessageContext;
 import org.wso2.carbon.apimgt.rest.api.util.utils.RestApiUtil;
 
 import javax.ws.rs.core.Response;
+import java.util.Map;
 
 public class SequenceApiServiceImpl implements SequenceApiService {
 
     private static final Log log = LogFactory.getLog(SequenceApiServiceImpl.class);
+    private final String SUPER_TENAT_DOMAIN = "carbon.super";
```
Comment (1akshitha) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/SequenceApiServiceImpl.java:41]: Point to the constant in APIConstants.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/SequenceApiServiceImpl.java [hunk 6]
```diff
@@ -27,22 +27,31 @@
...
 
-    public Response sequenceGet(String apiName, String label, String apiId, MessageContext messageContext) {
-
+    public Response sequenceGet(String apiName, String version , String tenantDomain, MessageContext messageContext) {
         InMemoryAPIDeployer inMemoryApiDeployer = new InMemoryAPIDeployer();
+        if (tenantDomain == null){
+            tenantDomain =SUPER_TENAT_DOMAIN;
```
Comment (1akshitha) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/SequenceApiServiceImpl.java:46]: Formatting error

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/UndeployApiApiServiceImpl.java [hunk 7]
```diff
@@ -20,16 +20,26 @@
...
 public class UndeployApiApiServiceImpl implements UndeployApiApiService {
+    private final String SUPER_TENAT_DOMAIN = "carbon.super";
 
-    public Response undeployApiPost(String apiName, String label, String apiId, MessageContext messageContext) {
+    public Response undeployApiPost(String apiName, String version , String tenantDomain,
+            MessageContext messageContext) {
 
         InMemoryAPIDeployer inMemoryApiDeployer = new InMemoryAPIDeployer();
+        if (tenantDomain == null){
+            tenantDomain =SUPER_TENAT_DOMAIN;
```
Comment (1akshitha) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/UndeployApiApiServiceImpl.java:38]: Formatting error

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml [hunk 8]
```diff
@@ -71,28 +71,32 @@ paths:
       security:
         - OAuth2Security:
             - apim:api_publish
-      x-wso2-curl: "curl -k -H \"Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8\" -X POST \"https://127.0.0.1:9443/api/am/gateway/v1/deploy-api?apiId=890a4f4d-09eb-4877-a323-57f6ce2ed79b&apiName=PizzaAPI&label=Food\""
+      x-wso2-curl: "curl -k -H \"Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8\" -X POST \"https://127.0.0.1:9443/api/am/gateway/v1/deploy-api?apiName=PizzaAPI&version=1.0&tenantDomain=registry.com\""
       x-wso2-request: |
-        POST https://127.0.0.1:9443/api/am/gateway/v1/deploy-api?apiId=890a4f4d-09eb-4877-a323-57f6ce2ed79b&apiName=PizzaAPI&label=Food
+        POST https://127.0.0.1:9443/api/am/gateway/v1/deploy-api?apiName=PizzaAPI&version=1.0&tenantDomain=registry.com
```
Comment (1akshitha) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml:76]: Shall we change this resource to redeploy?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml [hunk 9]
```diff
@@ -71,28 +71,32 @@ paths:
...
           description: |
             Name of the API
           in: query
           type: string
           required: true
-        - name: label
+        - name: version
           description: |
-            Label of the API Gateway
+            version of the API Gateway
```
Comment (1akshitha) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml:91]: The version of the API right?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml [hunk 10]
```diff
@@ -71,28 +71,32 @@ paths:
...
+      x-wso2-curl: "curl -k -H \"Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8\" -X POST \"https://127.0.0.1:9443/api/am/gateway/v1/deploy-api?apiName=PizzaAPI&version=1.0&tenantDomain=registry.com\""
       x-wso2-request: |
-        POST https://127.0.0.1:9443/api/am/gateway/v1/deploy-api?apiId=890a4f4d-09eb-4877-a323-57f6ce2ed79b&apiName=PizzaAPI&label=Food
+        POST https://127.0.0.1:9443/api/am/gateway/v1/deploy-api?apiName=PizzaAPI&version=1.0&tenantDomain=registry.com
         Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8
       x-wso2-response: "HTTP/1.1 200 OK"
       summary: Deploy the API in the gateway
       description: |
-        This operation is used to deploy and API in the gateway.
+        This operation is used to deploy an API in the gateway.
```
Comment (1akshitha) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml:81]: Better to mention if tenant domain is not given, super tenant get chosen

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml [hunk 11]
```diff
@@ -71,28 +71,32 @@ paths:
...
           description: |
-            Label of the API Gateway
+            version of the API Gateway
           in: query
           type: string
           required: true
-        - $ref: '#/parameters/apiId-Q'
+        - name: tenantDomain
+          description: |
+            tenantDomain of the API Gateway
```
Comment (1akshitha) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml:97]: "tenantDomain" should be "Tenant domain".

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml [hunk 12]
```diff
@@ -152,13 +156,17 @@ paths:
...
```
Comment (1akshitha) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml:167]: Tenant domain

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml [hunk 13]
```diff
@@ -288,29 +296,33 @@ paths:
           in: query
           type: string
           required: true
-        - name: label
+        - name: version
           description: |
-            Label of the API Gateway
+            version of the API Gateway
```
Comment (1akshitha) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml:301]: version of the API

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml [hunk 14]
```diff
@@ -288,29 +296,33 @@ paths:
...
           in: query
           type: string
           required: true
-        - $ref: '#/parameters/apiId-Q'
+        - name: tenantDomain
+          description: |
+            tenantDomain of the API Gateway
```
Comment (1akshitha) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml:307]: Tenant domain

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml [hunk 15]
```diff
@@ -420,16 +435,20 @@ paths:
...
```
Comment (1akshitha) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml:446]: Tenant domain

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml [hunk 16]
```diff
@@ -420,16 +435,20 @@ paths:
           in: query
           type: string
           required: true
-        - name: label
+        - name: version
           description: |
-            Label of the API Gateway
+            version of the API Gateway
```
Comment (1akshitha) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml:440]: version of API

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml (hunk 8 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml:76]: https://github.com/wso2/carbon-apimgt/pull/9009/commits/b99a273017722380eaa25c62f86587c522498b9a#diff-3628fb185e745131f3d5bb8e72370524R23

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml (hunk 10 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml]: https://github.com/wso2/carbon-apimgt/pull/9009/commits/b99a273017722380eaa25c62f86587c522498b9a#diff-3628fb185e745131f3d5bb8e72370524R27

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml (hunk 11 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml]: https://github.com/wso2/carbon-apimgt/pull/9009/commits/b99a273017722380eaa25c62f86587c522498b9a#diff-3628fb185e745131f3d5bb8e72370524R107

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml (hunk 16 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml]: https://github.com/wso2/carbon-apimgt/pull/9009/commits/b99a273017722380eaa25c62f86587c522498b9a

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml (hunk 15 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml]: https://github.com/wso2/carbon-apimgt/pull/9009/commits/b99a273017722380eaa25c62f86587c522498b9a

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml (hunk 14 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml]: https://github.com/wso2/carbon-apimgt/pull/9009/commits/b99a273017722380eaa25c62f86587c522498b9a

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml (hunk 13 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml]: https://github.com/wso2/carbon-apimgt/pull/9009/commits/b99a273017722380eaa25c62f86587c522498b9a

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml (hunk 12 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml]: https://github.com/wso2/carbon-apimgt/pull/9009/commits/b99a273017722380eaa25c62f86587c522498b9a

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml (hunk 9 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml]: https://github.com/wso2/carbon-apimgt/pull/9009/commits/b99a273017722380eaa25c62f86587c522498b9a

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/UndeployApiApiServiceImpl.java (hunk 7 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/UndeployApiApiServiceImpl.java]: https://github.com/wso2/carbon-apimgt/pull/9009/commits/b99a273017722380eaa25c62f86587c522498b9a

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/SequenceApiServiceImpl.java (hunk 6 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/SequenceApiServiceImpl.java]: https://github.com/wso2/carbon-apimgt/pull/9009/commits/b99a273017722380eaa25c62f86587c522498b9a

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/SequenceApiServiceImpl.java (hunk 5 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/SequenceApiServiceImpl.java]: https://github.com/wso2/carbon-apimgt/pull/9009/commits/b99a273017722380eaa25c62f86587c522498b9a

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/LocalEntryApiServiceImpl.java (hunk 4 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/LocalEntryApiServiceImpl.java]: https://github.com/wso2/carbon-apimgt/pull/9009/commits/b99a273017722380eaa25c62f86587c522498b9a

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/EndPointsApiServiceImpl.java (hunk 3 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/EndPointsApiServiceImpl.java]: https://github.com/wso2/carbon-apimgt/pull/9009/commits/b99a273017722380eaa25c62f86587c522498b9a

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/DeployApiApiServiceImpl.java (hunk 2 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/DeployApiApiServiceImpl.java]: https://github.com/wso2/carbon-apimgt/pull/9009/commits/b99a273017722380eaa25c62f86587c522498b9a

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml (hunk 1 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml]: https://github.com/wso2/carbon-apimgt/pull/9009/commits/b99a273017722380eaa25c62f86587c522498b9a

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml (hunk 8 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml]: https://github.com/wso2/carbon-apimgt/pull/9009/commits/7bc5f9c2d7bdf813d244a41d3a8c078bde6e54c8

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/EndPointsApiServiceImpl.java [hunk 17]
```diff
@@ -26,22 +26,33 @@
...
+            tenantDomain = APIConstants.SUPER_TENANT_DOMAIN;
+        }
+
+        Map<String, String> apiAttributes = inMemoryApiDeployer.getGatewayAPIAttributes(apiName, version, tenantDomain);
+        String apiId = apiAttributes.get(APIConstants.GatewayArtifactSynchronizer.API_ID);
+        String label = apiAttributes.get(APIConstants.GatewayArtifactSynchronizer.LABEL);
+
         GatewayAPIDTO gatewayAPIDTO = inMemoryApiDeployer.getAPIArtifact(apiId, label);
         JSONObject responseObj = new JSONObject();
+
```
Comment (Shehanir) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/EndPointsApiServiceImpl.java:55]: Do we need extra lines?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/RedeployApiApiServiceImpl.java [hunk 18]
```diff
@@ -0,0 +1,49 @@
+package org.wso2.carbon.apimgt.rest.api.gateway.v1.impl;
```
Comment (renuka-fernando) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/RedeployApiApiServiceImpl.java:1]: Shall we add license headers?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/RedeployApiApiServiceImpl.java [hunk 19]
```diff
@@ -0,0 +1,49 @@
...
+
+import java.util.List;
+
+import java.io.InputStream;
+import java.util.Map;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
+
+
```
Comment (chamindias) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/RedeployApiApiServiceImpl.java:23]: Please remove the unnecessary new lines.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/RedeployApiApiServiceImpl.java [hunk 20]
```diff
@@ -0,0 +1,49 @@
...
+            tenantDomain = APIConstants.SUPER_TENANT_DOMAIN;
+        }
+        Map<String, String> apiAttributes = inMemoryApiDeployer.getGatewayAPIAttributes(apiName, version, tenantDomain);
+        String apiId = apiAttributes.get(APIConstants.GatewayArtifactSynchronizer.API_ID);
+        String label = apiAttributes.get(APIConstants.GatewayArtifactSynchronizer.LABEL);
+        boolean status = inMemoryApiDeployer.deployAPI(apiId, label);
+
+        JSONObject responseObj = new JSONObject();
+        if (status) {
+            responseObj.put("Message", "Success");
```
Comment (chamindias) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/RedeployApiApiServiceImpl.java:40]: Can't we use a constant here?

Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/RedeployApiApiServiceImpl.java:40]: Response Changed to **apiName + " redeployed successfully in the Gateway"** in the latest version

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/RedeployApiApiServiceImpl.java (hunk 19 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/RedeployApiApiServiceImpl.java:23]: https://github.com/wso2/carbon-apimgt/pull/9546/commits/dfbd644d35add8643bb9aa652caba4e12f511b5b

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/RedeployApiApiServiceImpl.java (hunk 18 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/RedeployApiApiServiceImpl.java:1]: https://github.com/wso2/carbon-apimgt/pull/9546/commits/dfbd644d35add8643bb9aa652caba4e12f511b5b

**Pull request merged**